### ISSUE TITLE
High CPU usage during bootstrapping if broker never accepts tcp connection

### ISSUE THREAD

**Issue by JochenABC** [dpkp/kafka-python#1166]
(This issue has similarities with https://github.com/dpkp/kafka-python/issues/1164, however IMHO the root cause seems to be different and it might need a different fix. That is why I'm opening another one.)

### Problem

When inserting an iptables rule to drop all traffic to port 9092:

```
sudo iptables -I INPUT -p tcp --dport 9092 -j DROP
```

... and then running this python script:

```python
import kafka

if __name__ == '__main__':
    kafka.client.KafkaClient(bootstrap_servers='127.0.0.1:9092')
```

... it will draw 100% cpu for 40 seconds (connection timeout reached)

#### Expected behaviour:

Script draws a reasonable amount of CPU

#### Actual behaviour:

Script draws as much cpu as it gets.

### Additional information

When enabling debug output for kafka module I get:

```
Bootstrapping cluster metadata from [('127.0.0.1', 9092, <AddressFamily.AF_INET: 2>)]
Attempting to bootstrap via node at 127.0.0.1:9092
<BrokerConnection node_id=bootstrap host=127.0.0.1/127.0.0.1 port=9092>: creating new socket
<BrokerConnection node_id=bootstrap host=127.0.0.1/127.0.0.1 port=9092>: setting socket option (6, 1, 1)
<BrokerConnection node_id=bootstrap host=127.0.0.1/127.0.0.1 port=9092>: connecting to 127.0.0.1:9092

---> hangs until connection timeout is reached. Then:

Connection attempt to <BrokerConnection node_id=bootstrap host=127.0.0.1/127.0.0.1 port=9092> timed out
<BrokerConnection node_id=bootstrap host=127.0.0.1/127.0.0.1 port=9092>: Closing connection. ConnectionError: timeout
<BrokerConnection node_id=bootstrap host=127.0.0.1/127.0.0.1 port=9092>: reconnect backoff 0.050213043774345226 after 1 failures
Unable to bootstrap from [('127.0.0.1', 9092, <AddressFamily.AF_INET: 2>)]
Traceback (most recent call last):
  File "bug.py", line 9, in <module>
    kafka.client.KafkaClient(bootstrap_servers='127.0.0.1:9092')
  File "/home/jtuchbreiter/venv/app3.6/lib/python3.6/site-packages/kafka/client_async.py", line 220, in __init__
    self.config['api_version'] = self.check_version(timeout=check_timeout)
  File "/home/jtuchbreiter/venv/app3.6/lib/python3.6/site-packages/kafka/client_async.py", line 840, in check_version
    raise Errors.NoBrokersAvailable()
kafka.errors.NoBrokersAvailable: NoBrokersAvailable

```

### Why is this case relevant?

This will happen when the machine running kafka server is down  e,g. for reboot / hardware broken.

### Version used

kafka-python module: Latest version from master (`git+https://github.com/dpkp/kafka-python@3ff3d75004f94fd55fa089297d3e2376e33ccda7`)

**Comment by JochenABC** [comment]
After applying the following patch CPU usage gets lower. While the patch is probably not usable it might indicate where the problem lies:

```
diff --git a/kafka/client_async.py b/kafka/client_async.py
index 612349e..55e4746 100644
--- a/kafka/client_async.py
+++ b/kafka/client_async.py
@@ -236,6 +236,7 @@ class KafkaClient(object):
             bootstrap.connect()
             while bootstrap.connecting():
                 bootstrap.connect()
+                time.sleep(0.1)
             if not bootstrap.connected():
                 bootstrap.close()
                 continue
```

**Comment by dpkp** [comment]
Fixed in PR #1175
